src/rebuild/srv.c

```diff
@@ -1069,6 +1069,53 @@ rebuild_task_destroy(struct rebuild_task *task)
 	D_FREE(task);
 }
 
+/* Try merge the tasks to the current task.
+ *
+ * return 1 means merging the rebuild target to other task.
+ * return 0 means these targets can not merge.
+ *
+ **/
+static int
+rebuild_try_merge_tgts(const uuid_t pool_uuid, uint32_t map_ver,
+		       struct pool_target_id_list *tgts_failed)
+{
+	struct rebuild_task *task;
+	struct rebuild_task *found = NULL;
+	int		    rc;
+
+	d_list_for_each_entry(task, &rebuild_gst.rg_queue_list,
+			      dst_list) {
+		if (uuid_compare(task->dst_pool_uuid, pool_uuid) == 0) {
+			found = task;
+			break;
+		}
+	}
+
+	if (found == NULL)
+		return 0;
+
+	D_DEBUG(DB_REBUILD, "("DF_UUID" ver=%u) id %u merge to task %p\n",
+		DP_UUID(pool_uuid), map_ver,
+		tgts_failed->pti_ids[0].pti_id, task);
+
+	/* Merge the failed ranks to existing rebuild task */
+	rc = pool_target_id_list_merge(&task->dst_tgts, tgts_failed);
+	if (rc)
+		return rc;
+
+	if (task->dst_map_ver < map_ver) {
```

Comment:
How can this happen? I thought we do rebuild in pool map version order, no?

Do we rebuild failed target in fseq order or rebuild them in parallel? I think current placement code still assume target with lower fseq DOWNOUT prior to target with higher fseq.

Reply:
this will happen when we add new targets to the rebuild task which is not being rebuild yet. Note: this function will be called when schedule new rebuild target as well.